src/api/movies.ts

```diff
@@ -1,11 +1,14 @@
 import { defaultClient } from '@/api/client';
+import type { Movie } from '@/types/movie';
 
-export const getAllMovies = async () => {
+export const getAllMovies = async (): Promise<Movie[]> => {
```

Comment:
I think we could as well do it by not declaring the return type here, but using the client like so (similarly in the case below):

```ts
export const getAllMovies = async () => {
  const { data } = await defaultClient.get<Movie[]>('/movies'); //<- this will mark the data returned as Movie[] type
  return data;
};
```

Reply:
I will definitely do that, but could You just explain a bit why such approach is better than typing the return? :smile:
Is it the fact that in this case the function can catch a mismatched type earlier?

Reply:
First, we get the data typed properly (in case we would do something more here, like mapping, transforming the data) and then the type is inferred which is the desired behavior - to do less and leave as much as possible to TypeScript 😄

Reply:
Ok thanks a lot! 
Now it's clear and I'll try to follow this pattern 👍